CI/wininstaller/installer.iss

```diff
@@ -0,0 +1,733 @@
+  // Return the modified user docs folder path
+  Result := UserAppdataPath;
+end;
+
+
+function GetUserDesktopFolder(Default: String): String;
+var
+  UserDesktopPath: String;
+  OriginalUserName: String;
+  CurrentSessionUserName: String;
+begin
+  // Retrieve the current username from the session
+  CurrentSessionUserName := '\' + GlobalUserName + '\';
+
+  // Retrieve the original username
+  OriginalUserName := '\' + GetUserNameString + '\';
+
+  // Expand the {userdesktop} constant
+  UserDesktopPath := ExpandConstant('{userdesktop}');
+
+  // Replace the original username with the current session username
+  StringChangeEx(UserDesktopPath, OriginalUserName, CurrentSessionUserName, True);
+
+  // Return the modified user desktop folder path
+  Result := UserDesktopPath;
+end;
+
+
+function InitializeSetup(): Boolean;
+begin
+  // Initialize the global variable during setup
+  GlobalUserName := GetCurrentSessionUserName();
+  GlobalUserDocsFolder := GetUserDocsFolder();
+  GlobalUserAppdataFolder := GetUserAppdataFolder();
+  
+  Result := True;
+end;
+
+
+function InitializeUninstall(): Boolean;
+begin
+  // Initialize the global variable during uninstall
+  GlobalUserName := GetCurrentSessionUserName();
+  GlobalUserDocsFolder := GetUserDocsFolder();
+  GlobalUserAppdataFolder := GetUserAppdataFolder();
+  
+  Result := True;
+end;
+
+
+procedure InitializeWizard();
+var
+  InstallPath, VCMIFolder, MapsFolder, DataFolder, Mp3Folder: String;
+begin
+  // Check if the application is already installed
+  IsUpgrade := RegQueryStringValue(HKCU, 'Software\VCMI', 'InstallPath', InstallPath);
+  if not IsUpgrade then
+  begin
+    // Create the install mode selection page only if it's not an upgrade
+    InstallModePage := CreateInputOptionPage(
+      wpWelcome,
+      ExpandConstant('{cm:SelectSetupInstallModeTitle}'),
+      ExpandConstant('{cm:SelectSetupInstallModeDesc}'),
+      ExpandConstant('{cm:SelectSetupInstallModeSubTitle}'),
+      True, False
+    );
+    
+    // Option 0
+    InstallModePage.Add(ExpandConstant(#13#10 + '  {cm:InstallForAllUsers}' + #13#10 + '   • {cm:InstallForAllUsers1}' + #13#10 + #13#10));
+    // Option 1
+    InstallModePage.Add(ExpandConstant(#13#10 + '  {cm:InstallForMeOnly}' + #13#10  +  '   • {cm:InstallForMeOnly1}' + #13#10 + '   • {cm:InstallForMeOnly2}' + #13#10));
+
+    if IsAdmin then
+    begin
+      // Default to "All Users"
+      InstallModePage.SelectedValueIndex := 0;
+    end
+    else
+    begin
+      // Default to "Me Only"
+      InstallModePage.SelectedValueIndex := 1;
+
+      // Disable the first option ("Install for All Users") for non-admins
+      InstallModePage.CheckListBox.ItemEnabled[0] := False;
+
+      // Force a redraw of the CheckListBox to fix appearance
+      InstallModePage.CheckListBox.Invalidate();
+    end;
+  end;
+ 
+  // Check for Heroes 3 installation paths
+  Heroes3Path := '';
+  Heroes3Path := RegistryQueryPath('SOFTWARE\GOG.com\Games\1207658787', 'path');
```

Comment:
does this variable become global? I see it's used in another function.

Reply:
Yes, it becomes global